test/promise/all.js

```diff
@@ -168,8 +169,10 @@ describe('Promise.all', function () {
           }, reject);
         });
       } else {
-        Promise.call(this, resolver);
+        self = new Promise(resolver);
       }
+      Object.setPrototypeOf(self, P.prototype);
```

Comment:
Please make sure this test bails out early if `Object.setPrototypeOf` is not shimmed in the current environment.